new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonUtils.java

```diff
@@ -0,0 +1,655 @@
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info(
+                "Waiting for NGINX to be ready in namespace {0} (elapsed time {1}ms, remaining time {2}ms)",
+                nginxNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(assertDoesNotThrow(() -> isNginxReady(nginxNamespace), "isNginxReady failed with ApiException"));
+
+    return nginxHelmParams;
+  }
+
+  /**
+   * Create a domain in the specified namespace and wait up to 5 minutes until the domain exists.
+   *
+   * @param domain the oracle.weblogic.domain.Domain object to create domain custom resource
+   * @param domainNamespace namespace in which the domain will be created
+   */
+  public static void createDomainAndVerify(Domain domain, String domainNamespace) {
+
+    // create the domain CR
+    assertNotNull(domain, "domain is null");
+    assertNotNull(domain.getSpec(), "domain spec is null");
+    String domainUid = domain.getSpec().getDomainUid();
+
+    logger.info("Creating domain custom resource for domainUid {0} in namespace {1}",
+        domainUid, domainNamespace);
+    assertTrue(assertDoesNotThrow(() -> createDomainCustomResource(domain),
+        String.format("Create domain custom resource failed with ApiException for %s in namespace %s",
+            domainUid, domainNamespace)),
+        String.format("Create domain custom resource failed with ApiException for %s in namespace %s",
+            domainUid, domainNamespace));
+
+    // wait for the domain to exist
+    logger.info("Checking for domain custom resource in namespace {0}", domainNamespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for domain {0} to be created in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                domainUid,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainExists(domainUid, DOMAIN_VERSION, domainNamespace));
+  }
+
+  /**
+   * Create an ingress for the domain with domainUid in the specified namespace.
+   *
+   * @param domainUid WebLogic domainUid which is backend to the ingress to be created
+   * @param domainNamespace WebLogic domain namespace in which the domain exists
+   * @param managedServerPort port number of WebLogic domain managed servers
+   * @param clusterNames list of the WebLogic domain cluster names in the domain
+   * @return list of ingress hosts
+   */
+  public static List<String> createIngressForDomainAndVerify(String domainUid,
+                                                             String domainNamespace,
+                                                             int managedServerPort,
+                                                             List<String> clusterNames) {
```

Comment:
you assume all of the clusters have the same listen port

Reply:
fixed